core/src/main/java/org/hibernate/ogm/dialect/multiget/spi/MultigetGridDialect.java

```diff
@@ -10,24 +10,26 @@
 
 import org.hibernate.ogm.dialect.spi.GridDialect;
 import org.hibernate.ogm.dialect.spi.TupleContext;
-import org.hibernate.ogm.model.key.spi.EntityKey;
+import org.hibernate.ogm.model.key.spi.EntityKeyMetadata;
 import org.hibernate.ogm.model.spi.Tuple;
 
 /**
  * A {@link GridDialect} facet representing dialects that can load several objects in one datastore operation.
  *
  * @author Emmanuel Bernard emmanuel@hibernate.org
+ * @author Davide D'Alto
  */
 public interface MultigetGridDialect extends GridDialect {
 
 	/**
-	 * Return the list of tuples with the given columns for a given list of keys.
-	 * The tuples must be returned in the same order as the keys.
+	 * Returns the list of tuples for the given {@link EntityKeyMetadata} and key columns values.
+	 * The tuples must be returned in the same order as the key columns values.
 	 * If a key has no matching record, set null to the list entry.
 	 *
-	 * @param keys The array of tuple identifier
+	 * @param metadata the {@link EntityKeyMetadata} of the expected tuples
+	 * @param values the values of the key columns, each row of the matrix contains one key values
 	 * @param tupleContext Contains additional information that might be used to create the tuples
-	 * @return the list of tuples identified by the keys
+	 * @return the list of tuples identified by the key values
 	 */
-	List<Tuple> getTuples(EntityKey[] keys, TupleContext tupleContext);
+	List<Tuple> getTuples(EntityKeyMetadata metadata, Object[][] values, TupleContext tupleContext);
```

Comment:
But the explicitly passed EKM makes it apparent that all keys are of that same type. Passing data around as `Object[][]` makes it just unnecessarily hard to understand the stuff, that's what classes are for :) Also that signature does not make clear anymore that this is about entity keys at all, it could just be anything.